src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4976,6 +4976,11 @@ def set_vm_applications(cmd, vm_name, resource_group_name, application_version_i
             if over_ride or over_ride.lower() != 'null':
                 vm.application_profile.gallery_applications[index].configuration_reference = over_ride
             index += 1
+    
+    if treat_deployment_as_failure:
+        index = 0
+        for treat_as_failure in treat_deployment_as_failure:
+            vm.application_profile.gallery_applications[index].treat_failure_as_deployment_failure = treat_as_failure
```

Comment:
The value of index needs to increase, right?
```suggestion
        index = 0
        for treat_as_failure in treat_deployment_as_failure:
            vm.application_profile.gallery_applications[index].treat_failure_as_deployment_failure = treat_as_failure
            index = index + 1
```

Reply:
fixed